virtool/api/response.py

```diff
@@ -26,27 +37,72 @@ def json_response(
 
 
 class InsufficientRights(HTTPForbidden):
+    """
+    Reusable exception for returning an HTTP status `403` (Insufficient Rights) message
+    to the client.
+    """
+
     def __init__(self, message="Insufficient rights"):
         super().__init__(text=message, reason="insufficient_rights")
 
 
 class NotFound(HTTPNotFound):
+    """
+    Reusable exception for returning an HTTP status `404` (Not Found) message to the
+    client.
+    """
+
     def __init__(self, message="Not found"):
         super().__init__(text=message, reason="not_found")
 
 
 class EmptyRequest(HTTPUnprocessableEntity):
+    """
+    Reusable exception for returning an HTTP status `422` (Empty Request) message to the
+    client.
+    """
+
     def __init__(self, message="Empty request"):
         super().__init__(text=message, reason="empty_request")
 
 
 class InvalidQuery(HTTPUnprocessableEntity):
+    """
+    Reusable exception for returning an HTTP status `422` (Invalid Query) message to the
+    client.
+    """
+
     def __init__(self, errors, message="Invalid query"):
         super().__init__(text=message, reason="invalid_query")
         self.errors = errors
 
 
 class InvalidInput(HTTPUnprocessableEntity):
+    """
+    Reusable exception for returning an HTTP status `422` (Invalid Input) message to the
+    client.
+    """
+
     def __init__(self, errors, message="Invalid input"):
         super().__init__(text=message, reason="invalid_input")
         self.errors = errors
+
+
+@web.middleware
+async def middleware(req: web.Request, handler: Callable):
```

Comment:
Use `from` imports where this is imported.
```suggestion
async def error_middleware(req: web.Request, handler: Callable):
```